### DISCUSSION TITLE
[Feature] Option to preserve (or don't) favorite photo status

### DISCUSSION THREAD

**Discussion by sizzleweet** [immich-app/immich#4061] [Feature Request]
### The feature

Hey all,

For iOS, I favorite all the photos that I deem as "keepers", and then have immich use that album for backing up.

In the app then, all the photos are already favorited.  I use my favorites album as a method for culling because it's so easy to tap the little heart icon on my phone.

But I don't want every single photo on immich tagged as favorites.  I want to pick the cream of the crop for those from my uploads.  So it'd be nice to have an option to preserve or not preserve the favorite status from the mobile upload.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #1669

**Reply by jrasm91** [reply]
I'm not sure if this is a duplicate or not. In the mentioned example the current app works like this:
- User selects "Favorites" album for backup in iOS
- User favorites a photo (becomes apart of the "Favorites" album on ios)
- Photo is uploaded to Immich with the property `isFavorite: true`
- All future changes to favorite status on the phone or in the web are unrelated/unsynced

I think the linked issue is talking about the last point, while OP is asking about the second to last point. Right now the status is in-sync _on upload_ and may get out of sync later. OP would like for it to _not_ be in sync, even on initial upload.

**Reply by sizzleweet** [reply]
Yes, this is exactly what I'm trying to say.

I'd like to be able to decide whether or not the "isFavorite" flag is preserved during import.

I want to be able to sync a picture from the favorites album and have it NOT be favorited in immich.

**Reply by jrasm91** [reply]
This seems like a great mobile-only setting. Something like `Favorite Status`: `true`, `false`, `match os`.

**Comment by aryanarora0303** [comment]
Following up with a concrete proposal — would a PR along these lines be welcome?

**Problem:** favorite status is only carried over at first upload (#5082). If a user favorites a photo on their phone *after* it has already been backed up, the change never reaches the server. For families replacing iCloud/Google Photos with Immich, favorites are a core curation habit — and people naturally keep favoriting in their native gallery after backup, where the change is silently lost.

**Proposal — opt-in, one-way re-sync (device → server):**
- New toggle in the mobile backup settings, **off by default**: "Keep favorites in sync".
- During a sync/backup pass, for assets that are already uploaded and **owned by the current user**, compare the device favorite flag to the server's `isFavorite`; if they differ, update the server.
- One-way only (device wins) and opt-in, so there is no conflict-resolution question and zero behavior change for anyone who leaves it off.
- Even more conservative flavor if clobbering server-set favorites is a concern: **additive-only** — the sync only ever *sets* favorites, never unsets them.

Happy to attempt the implementation in the mobile app if maintainers think this approach (or a variant) is acceptable — guidance on which flavor you'd prefer would be much appreciated before I start.
